lib/helpers.js

```diff
@@ -1,69 +1,97 @@
-const authToken = require('./auth-token');
-const User = require('../models/user');
-const Driver = require('../models/driver');
+const authToken = require('./auth-token')
+const User = require('../models/user')
+const Driver = require('../models/driver')
+const safeEquals = require('safe-compare')
 
-module.exports = {
-  requireAuthentication: async function requireAuthentication(req, res, next) {
-    try {
-      let header = req.header('Authorization');
-      let token = header.split(' ')[1];
-      var payload = authToken.decode(token);
+const genericErrorMessage = message => ({ errors: [{ message }] })
+const extractToken = req => req.header('Authorization').split(' ')[1]
+const verifyAndDecodeTokenPayload = (req, token) => (req.payload = authToken.decode(token))
+
+async function withErrorHandler (res, fn) {
```

Comment:
This is a good solution for the async await error handling 👌

Reply:
And I need to say that I have an issue with the space between the function name and the parentheses, I think we need a linter

Reply:
And semicolons, I know that they are not necessary but we like them. 😅

Reply:
ahh ok, I will disable my linter.
I tend to use https://standardjs.com/